framework/src/actions/SetupMeshAction.C

```diff
@@ -84,6 +84,15 @@ SetupMeshAction::validParams()
 
   params.addParam<unsigned int>(
       "uniform_refine", 0, "Specify the level of uniform refinement applied to the initial mesh");
+
+  params.addParam<bool>(
+      "uniform_refine_remote_deletion",
+      true,
+      "If it is false, remote deletion will be avoided for efficiency of uniform refinements. "
+      "This flag has no impact on a replicated mesh. For a distributed mesh, the remote elements "
+      "are already deleted, "
+      "and the mesh is already distributed properly, before uniform refinements.");
```

Comment:
The explanation here needs serious work.

You're turning off *both* repartitioning *and* remote element deletion here, yes?  Disabling repartitioning can mean a greater load imbalance on the fine mesh if the coarse mesh wasn't partitioned evenly enough, so we definitely don't want to omit mention of that effect.  If you start with `N_elem >> N_proc` on the coarse mesh then of course you won't care, but it's not unreasonable for users to start from extremely coarse meshes, even when they intend to only do fine mesh solves, if the domain is simple enough.

Also, "the remote elements are already deleted" is not true.  If you start with N point neighbor elements geometrically ghosted on a d-dimensional mesh, and you refine it r times, you'll end up wanting approximately `N*2^{r*(d-1)}` elements ghosted, but you'll have `N*2^{r*d}` elements still geometrically ghosted instead.  The difference might not be enough to care about, but it's a difference.  Maybe just say "most of the remote elements are already deleted"?

Reply:
> "the remote elements are already deleted" is not true

I meant that the remote elements of the original mesh were deleted. The extra remote elements because of uniform refinements will take some memory, but it is way better than a replicated mesh. For example, if my default geometry ghosting layer  is one layer, and I will have have only 4 ghosting layers after two levels of refinements. It still looks good for me. 

Thanks for comments. I will try to rephrase statements. Might be even worth documenting